cmd/stack_create.go

```diff
@@ -0,0 +1,190 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+type stackCreateCommandConfig struct {
+	*RootCommandConfig
+	copy string
+}
+
+func newStackCreateCmd(rootConfig *RootCommandConfig) *cobra.Command {
+	config := &stackCreateCommandConfig{RootCommandConfig: rootConfig}
+
+	var stackCmd = &cobra.Command{
+		Use:   "create",
+		Short: "Create a new stack as a copy of an existing stack",
+		Long: `This command will create a new stack as a copy of an existing sample stack in the current directory that has the structure of an Appsody stack.
+        
+		If a copy flag is specified, stack create command will create a new stack as a copy of that existing stack.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			if len(args) < 1 {
+				return errors.New("Stack create command should have a stack name. Run `appsody stack create <name>` to create one")
+			}
+
+			stack := args[0]
+
+			exists, _ := Exists(stack)
+
+			if exists {
+				return errors.New("This stack named " + stack + " already exists")
+			}
+
+			err := downloadFolderToDisk("https://github.com/appsody/stacks/archive/master.zip", getHome(rootConfig)+"/extract/repo.zip")
+			if err != nil {
+				return err
+			}
+
+			if config.copy != "" {
+				repoIndex := strings.Index(config.copy, "/")
+
+				copiedStack := config.copy[repoIndex+1:]
+
+				valid, unzipErr := unzip(getHome(rootConfig)+"/extract/repo.zip", stack, config.copy)
+
+				if unzipErr != nil {
+					return unzipErr
+				}
+
+				if !valid {
+					return errors.Errorf("This is not a valid stack. Please specify any existing stack as <repo>/<stack>")
+				}
+				os.Remove(getHome(rootConfig) + "/extract/repo.zip")
+
+				err := os.Rename(stack+"/stacks-master/"+config.copy, copiedStack)
+				if err != nil {
+					return err
+				}
+
+				os.RemoveAll(stack)
+
+				err = os.Rename(copiedStack, stack)
+				if err != nil {
+					return err
+				}
+
+			} else {
+				valid, unzipErr := unzip(getHome(rootConfig)+"/extract/repo.zip", stack, "")
+
+				if unzipErr != nil {
+					return unzipErr
+				}
+
+				if !valid {
+					return errors.Errorf("This is not a valid stack. Please specify any existing stack as <repo>/<stack>")
+				}
+				os.Remove(getHome(rootConfig) + "/extract/repo.zip")
+
+				err := os.Rename(stack+"/stacks-master/samples/sample-stack/", "sample-stack")
+				if err != nil {
+					return err
+				}
+
+				os.RemoveAll(stack)
+
+				err = os.Rename("sample-stack", stack)
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		},
+	}
+	stackCmd.PersistentFlags().StringVar(&config.copy, "copy", "", "Copy existing stack")
```

Comment:
Set the default here to `"sample-stack"`, and then you don't need the big `else` block above to handle config.copy == "".